### PR TITLE
Created controllers, dtos and mappers for order, shopping cart and user

### PR DESCRIPTION
**Pull request by YuraMitriakhin**

### CODE REVIEW

**Review by IwalkInSilence — APPROVED** [review]

File: src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+
+    @Autowired
+    public AuthenticationController(AuthenticationService authenticationService) {
+        this.authenticationService = authenticationService;
+    }
+
+    @PostMapping("/register")
+    public void register(UserRequestDto userRequestDto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java]: how do we get this dto?

File: src/main/java/com/gmail/yuramitryahin/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    @PostMapping("/complete")
+    public void complete(@RequestParam Long userId) {
+        User user = userService.get(userId);
+        orderService.completeOrder(shoppingCartService.getByUser(user));
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> get(@RequestParam Long userId) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/controller/OrderController.java]:
```suggestion
    public List<OrderResponseDto> getHistory(@RequestParam Long userId) {
```

File: src/main/java/com/gmail/yuramitryahin/controller/ShoppingCartController.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        MovieSession movieSession = movieSessionService.get(movieSessionId);
+        shoppingCartService.addSession(movieSession, user);
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
+        User user = userService.get(userId);
+        return shoppingCartMapper.toDto(shoppingCartService.getByUser(user));
+    }
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/controller/ShoppingCartController.java]: redundant empty line

File: src/main/java/com/gmail/yuramitryahin/dao/MovieSessionDao.java [hunk 4]
```diff
@@ -12,4 +12,6 @@
     MovieSession update(MovieSession movieSession);
 
     void delete(Long id);
+
+    MovieSession get(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/dao/MovieSessionDao.java]: should we return Optional here?

File: src/main/java/com/gmail/yuramitryahin/dao/UserDao.java [hunk 5]
```diff
@@ -7,4 +7,6 @@
     User add(User user);
 
     Optional<User> findByEmail(String email);
+
+    User get(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/dao/UserDao.java]: same

File: src/main/java/com/gmail/yuramitryahin/model/dto/OrderResponseDto.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.NoArgsConstructor;
+
+@Builder
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketIds;
+    private LocalDateTime orderDate;
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/model/dto/OrderResponseDto.java:17]:
```suggestion
    private String orderDate;
```

File: src/main/java/com/gmail/yuramitryahin/model/dto/ShoppingCartResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class ShoppingCartResponseDto {
+    private long id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/model/dto/ShoppingCartResponseDto.java]:
```suggestion
    private Long id;
```

File: src/main/java/com/gmail/yuramitryahin/model/dto/UserResponseDto.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class UserResponseDto {
+    private long id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/gmail/yuramitryahin/model/dto/UserResponseDto.java]:
```suggestion
    private Long id;
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/gmail/yuramitryahin/model/dto/OrderResponseDto.java (hunk 6 above)
Comment (IwalkInSilence) [reply src/main/java/com/gmail/yuramitryahin/model/dto/OrderResponseDto.java:17]: Bohdan said that we better send LocalDateTime

File: src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java (hunk 1 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java]: what do you mean ?

File: src/main/java/com/gmail/yuramitryahin/controller/OrderController.java (hunk 2 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controller/OrderController.java]: fixed

File: src/main/java/com/gmail/yuramitryahin/controller/ShoppingCartController.java (hunk 3 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controller/ShoppingCartController.java]: fixed

File: src/main/java/com/gmail/yuramitryahin/dao/MovieSessionDao.java (hunk 4 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/dao/MovieSessionDao.java]: fixed

Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/dao/MovieSessionDao.java]: but if we do not find such an object, will we still have an exception?

File: src/main/java/com/gmail/yuramitryahin/dao/UserDao.java (hunk 5 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/dao/UserDao.java]: fixed

File: src/main/java/com/gmail/yuramitryahin/model/dto/ShoppingCartResponseDto.java (hunk 7 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/model/dto/ShoppingCartResponseDto.java]: fixed

File: src/main/java/com/gmail/yuramitryahin/model/dto/UserResponseDto.java (hunk 8 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/model/dto/UserResponseDto.java]: fixed

File: src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java]: how do you pass this dto to the method? have you tested it?

File: src/main/java/com/gmail/yuramitryahin/dao/MovieSessionDao.java (hunk 4 above)
Comment (KseniiaMakarova) [reply src/main/java/com/gmail/yuramitryahin/dao/MovieSessionDao.java]: sure - will we have an exception if we try to get value from empty Optional on the service level?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
good job
`register` method is yet to be fixed - see comments above

File: src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java (hunk 1 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java]: dto comes as json and parsed

Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java]: all working, I tested

File: src/main/java/com/gmail/yuramitryahin/dao/MovieSessionDao.java (hunk 4 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/dao/MovieSessionDao.java]: ok

File: src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java (hunk 1 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java]: in dto I have two fields email and password

Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controller/AuthenticationController.java]: FIXED

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**